src/storage.rs

```diff
@@ -1,15 +1,58 @@
 use std::fmt::Debug;
 
+use ahash::AHashMap;
 use alloy_primitives::{Address, Bytes, B256, U256};
 use revm::{
-    primitives::{AccountInfo, Bytecode},
+    db::PlainAccount,
+    primitives::{Account, AccountInfo, Bytecode},
     DatabaseRef,
 };
 
+/// An EVM account.
+// TODO: Flatten `AccountBasic` or more ideally, replace this with an Alloy type.
+// `AccountBasic` works for now as we're tightly tied to REVM types, hence
+// conversions between `AccountBasic` & `AccountInfo` are very convenient.
+#[derive(Debug, Default, Clone, PartialEq)]
+pub struct EvmAccount {
+    /// The account's basic information.
+    pub basic: AccountBasic,
+    /// The account's storage.
+    pub storage: AHashMap<U256, U256>,
+}
+
+impl From<PlainAccount> for EvmAccount {
+    fn from(account: PlainAccount) -> Self {
+        EvmAccount {
+            basic: account.info.into(),
+            storage: account.storage.into_iter().collect(),
+        }
+    }
+}
+
+impl EvmAccount {
```

Comment:
It's minor but it's cleaner to put this `impl` right below the `struct` definition with two `From` instances following aftewards.